### PR TITLE
#M10. Update Hackathon and Profile model with new fields. #91 ~ Auxfuse

### PR DESCRIPTION
**Pull request by auxfuse**
## Description

`Organisation` Model created inside accounts app, and Signup Form field of `organisation` added to capture User's Organisation on signup via `ModelChoiceField` field.

Hackathon Model addition with `organisation` field and foreign key to Model with same name, to capture the Parent Organisation of any created hackathon, small inclusion of allowing for a hackathon to not be hosted by an Organisation by allowing it to be left blank in the admin panel.

Hackathon Model added to with _status_ field, with values injected via `STATUS_TYPES_CHOICES` list imported via lists.py file in hackathon app.

Additional words includes Ammended navbar to include Admin Panel nav-item under My Account drop down-menu when authenticated user is superuser/admin. This improves flow of the site for bespoke user types instead of having to append the url of the browser to include `/admin` to navigate to same. Screenshot below.

## Pull request type
- [x] #M10. Update Hackathon and Profile model with new fields. #91
- [x] Feature with new Model, and update to existing models and forms.

## Testing

Fields added and manually tested via record creation.

Updated navigation tested with new inclusive link to admin panel, based on user type of superuser/admin. 

## Screenshots

*Screenshot depicting organisation field on signup (registration) of user:

image

* Screenshot showing creation of Organisation via Organisation model in Accounts app admin panel: 

image

* Screenshot showing creation of Hackathon with both new fields added:

image

* Screenshot showing update to Navigation items based on if user is authenticated as Administrator:

image

## Does this introduce a breaking change

- [x] No

### CODE REVIEW

File: accounts/models.py [hunk 1]
```diff
@@ -26,6 +36,14 @@ class CustomUser(AbstractUser):
         choices=LMS_MODULES_CHOICES
     )
 
+    organisation = models.ForeignKey(
+        Organisation,
+        on_delete=models.CASCADE,
+        related_name='user_organisation',
+        default='',
```
Comment (stefdworschak) [inline accounts/models.py:43]: That's only needed for Char or TextField and also you have `null=True` set as well.

File: hackathon/models.py [hunk 2]
```diff
@@ -2,6 +2,9 @@
 from django.utils import timezone
 
 from accounts.models import CustomUser as User
+from accounts.models import Organisation
+
```
Comment (stefdworschak) [inline hackathon/models.py:6]: No extra blank line needed here

**Review by stefdworschak — CHANGES_REQUESTED** [review]
@auxfuse overall the changes to the models etc. look good to me. However I am concerned about you deleting and changing a lot of the migrations. I would suggest to just add migrations changing the existing databases. If there are some conflicts with your database, I would delete the `db.sqlite3` file and run the migrations again. That way there should not be any conflicts.

File: accounts/models.py (hunk 1 above)
Comment (auxfuse) [reply accounts/models.py]: Removed `null=True` and refactored `default=''` to take in default ID of 1 based on Organisation Object display name. Seeding the data from the organisation.json file first to ensure this property works as expected.

https://stackoverflow.com/questions/9311996/setting-default-value-for-foreign-key-attribute/40813536#40813536 Resolution drafted from here with your help Stefan! Thank you for that.

File: hackathon/models.py (hunk 2 above)
Comment (auxfuse) [reply hackathon/models.py]: Removed additional blank line.

**Review by stefdworschak — APPROVED** [review]
LGTM

**Pull request merged**